Packs/QualysVulnerabilityManagement/Integrations/QualysVulnerabilityManagement/QualysVulnerabilityManagement_test.py

```diff
@@ -1,546 +1,606 @@
+        """
+        reponse = {}
+        res = is_empty_result(reponse)
+        assert res
+
+    def test_is_empty_xml_only_datetime(self):
+        """
+        Given
+            - A json parsed result from qualys
+        When
+            - result has only datetime key
+        Then
+            - return true since result has no content
+        """
+        response = {'DATETIME': 'sometime'}
+        res = is_empty_result(response)
+        assert res
+
+    def test_is_empty_xml_non_empty_result(self):
+        """
+        Given
+            - A json parsed result from qualys
+        When
+            - result has some keys
+        Then
+            - return false since result has content
+        """
+        response = {'IP_SET': {'IP': ['1.1.1.1']},
+                    'DATETIME': 'sometime'}
+        res = is_empty_result(response)
+        assert not res
+
+    def test_is_empty_xml_none_result(self):
+        """
+        Given
+            - A result from qualys
+        When
+            - result is None
+        Then
+            - return true
+        """
+        response = None
+        assert is_empty_result(response)
+
+
+class TestFormatAndValidateResponse:
+    def test_format_and_validate_response_proper_response(self):
+        """
+        Given
+            - raw xml response
+        When
+            - the response is valid
+        Then
+            - return the parsed response
+        """
+        raw_xml_response = '<?xml version="1.0" encoding="UTF-8" ?>'\
+                           '<!DOCTYPE SIMPLE_RETURN SYSTEM' \
+                           ' "https://qualysapi.qg2.apps.qualys.com/api/2.0/simple_return.dtd">' \
+                           '<SIMPLE_RETURN><RESPONSE>' \
+                           '<DATETIME>2021-03-24T15:40:23Z</DATETIME>' \
+                           '<TEXT>IPs successfully added to Vulnerability Management</TEXT>' \
+                           '</RESPONSE></SIMPLE_RETURN>'
+        raw_json_response = format_and_validate_response(raw_xml_response)
+        assert raw_json_response.get('SIMPLE_RETURN').get('RESPONSE')
+        assert not raw_json_response.get('CODE')
+
+    def test_format_and_validate_response_error_response(self):
+        """
+        Given
+            - raw xml response
+        When
+            - the response has an error code provided by qualys
+        Then
+            - raise a DemistoException
+        """
+        with pytest.raises(DemistoException):
+            raw_xml_response = '<?xml version="1.0" encoding="UTF-8" ?>' \
+                               '<!DOCTYPE SIMPLE_RETURN SYSTEM ' \
+                               '"https://qualysapi.qg2.apps.qualys.com/api/2.0/simple_return.dtd">' \
+                               '<SIMPLE_RETURN>' \
+                               '<RESPONSE><DATETIME>2021-03-24T16:35:44Z</DATETIME>' \
+                               '<CODE>1905</CODE><TEXT>IP(s) do not exist.</TEXT></RESPONSE></SIMPLE_RETURN>'
```

Comment:
Think about making it a global/class variable.

Reply:
Fixed